### PR TITLE
Use cli transfer

### PR DESCRIPTION
**Pull request by erickmartins**
This update reworks jax-omeroutils significantly to use omero-cli-transfer as the mechanism to import and annotate images. It also includes a fix for #24 .

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,5 @@
-zeroc-ice==3.6.5
-omero-py==5.10.0
-pandas==1.1.5
-numpy==1.19.5
-ezomero>=0.3.1
-openpyxl==3.0.9
\ No newline at end of file
+pandas>=1.1.5
+numpy>=1.22.0,<2.0.0
+ezomero>=2.1.0
```
Comment (govekk) [inline requirements.txt:3]: Conflicting dependencies: omero-cli-transer 0.7.3 requires ezomero 2.0.0, this requirements file requires ezomero>=2.1.0

File: jax_omeroutils/xml_editor.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    kv = MapAnnotation(**kwargs)
+    kvref = AnnotationRef(id=kv.id)
+    return kv, kvref
+
+
+def add_projects_datasets(ome, imp_json):
+    newome = copy.deepcopy(ome)
+    columns = imp_json['user_supplied_md']['file_metadata'][0].keys()
+    if 'project' not in columns:
+        return ome
```
Comment (govekk) [inline jax_omeroutils/xml_editor.py:39]: May want to also check for dataset in columns here. Likely doesn't actually matter because it won't reach this code without having both project and dataset columns given the other checks in other functions

File: jax_omeroutils/xml_editor.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                    img_id = ann.namespace
+                    for i in newome.images:
+                        if i.id == img_id:
+                            mmap = []
+                            for _key, _value in ann_dict.items():
+                                if _value:
+                                    mmap.append(M(k=_key, value=str(_value)))
+                                else:
+                                    mmap.append(M(k=_key, value=''))
+                            ann, annref = create_kv_and_ref(
```
Comment (govekk) [inline jax_omeroutils/xml_editor.py:122]: Would be easier to read this for loop if the `ann` variable wasn't overwritten. Could call this one `newann` or call the loop variable `filename_ann` to make it more clear we're looping through to find filenames in the CommentAnnotations

File: jax_omeroutils/xml_editor.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                    pl_id = ann.namespace
+                    for pl in newome.plates:
+                        if pl.id == pl_id:
+                            mmap = []
+                            for _key, _value in ann_dict.items():
+                                if _value:
+                                    mmap.append(M(k=_key, value=str(_value)))
+                                else:
+                                    mmap.append(M(k=_key, value=''))
+                            ann, annref = create_kv_and_ref(
```
Comment (govekk) [inline jax_omeroutils/xml_editor.py:162]: Same thing here: ann variable from the loop is overwritten, which makes it a bit hard to parse

**Review by govekk — CHANGES_REQUESTED** [review]
Looks good! Just a few small comments - mainly I'd like to change the variable names in the add_annotation functions so the loop variable isn't being overwritten

File: jax_omeroutils/xml_editor.py (hunk 4 above)
Comment (erickmartins) [reply jax_omeroutils/xml_editor.py:162]: these loops are horrendous regardless, but agreed

**Review by govekk — APPROVED** [review]
